src/main/webapp/creatives/maps/simid_maps.js

```diff
@@ -1,8 +1,44 @@
 /**
  * A sample SIMID ad that shows a map of nearby locations
  */
+const AdParamKeys = {
+  BUTTON_LABEL: 'buttonLabel',
+  SEARCH_QUERY: 'searchQuery',
+  MARKER: 'marker',
+}
+const DEFAULT_BUTTON_LABEL = "Location";
+
 class SimidMapCreative extends BaseSimidCreative {
   constructor() {
     super();
   }
+
+  /** @override */
+  onStart(eventData){
+    super.onStart(eventData);
+    const adParams = this.creativeData.adParameters;
+    const buttonLabel = JSON.parse(adParams)[AdParamKeys.BUTTON_LABEL]; 
+    if (buttonLabel == undefined) {
+      buttonLabel = DEFAULT_BUTTON_LABEL;
+    }
+    else {
+      buttonLabel = JSON.parse(adParams)[AdParamKeys.BUTTON_LABEL];
+    }
+    let searchQuery = JSON.parse(adParams)[AdParamKeys.SEARCH_QUERY];
+    //ToDo(juliareichel@): handle case where searchQuery is undefined
+    const marker = JSON.parse(adParams)[AdParamKeys.MARKER];;
+    this.specifyButtonFeatures_(buttonLabel);
+  }
+ 
+  /** @private */
+  specifyButtonFeatures_(buttonLabel) {
```

Comment:
optional nit.
You can use default parameters to handle setting the default value here.

`specifyButtonFeatures_(buttonLabel = DEFAULT_LABEL) {`

Reply:
Cool, never used this before. Changed this!